### PR TITLE
[Editor] Provide an element to render in the annotation layer after a freetext has been edited (bug 1890535)

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]
These changes may affect general annotation-rendering, please run *all* tests.

**Comment by Snuffleupagus** [comment]
Given the annotation-related changes here, this seems like something that shouldn't land just prior to a new release.

@timvandermeij This would be one more reason to make a new release ASAP, in my opinion.

File: src/display/annotation_layer.js [hunk 1]
```diff
@@ -197,6 +201,52 @@ class AnnotationElement {
     return AnnotationElement._hasPopupData(this.data);
   }
 
+  update(params) {
```
Comment (Snuffleupagus) [inline src/display/annotation_layer.js:204]: Given where/how this is being used, how about making that clearer in the name (to help future readers of the code)?
```suggestion
  updateEdited(params) {
```

File: src/display/annotation_layer.js [hunk 2]
```diff
@@ -197,6 +201,52 @@ class AnnotationElement {
...
+    };
+
+    const { rect } = params;
+
+    if (rect) {
+      this.#setRect(rect);
+    }
+  }
+
+  reset() {
```
Comment (Snuffleupagus) [inline src/display/annotation_layer.js:220]: As above.
```suggestion
  resetEdited() {
```

File: src/display/editor/editor.js [hunk 3]
```diff
@@ -1336,6 +1336,17 @@ class AnnotationEditor {
...
   }
 
+  /**
+   * Check if an existing annotation associated with this editor has been
+   * modified.
+   * @returns {boolean}
+   */
+  get hasBeenModified() {
+    return (
+      this.annotationElementId && (this.deleted || this.serialize() !== null)
```
Comment (Snuffleupagus) [inline src/display/editor/editor.js:1346]: As mentioned recently in another review comment, please take care to ensure that the return value is *consistent* in getters!
```suggestion
      !!this.annotationElementId && (this.deleted || this.serialize() !== null)
```

**Review by Snuffleupagus — COMMENTED** [review]
Leaving a few comments based on a cursory look.

File: src/display/annotation_layer.js [hunk 4]
```diff
@@ -197,6 +201,52 @@ class AnnotationElement {
...
+
+  resetEdited() {
+    if (!this.#updates) {
+      return;
+    }
+    this.#setRect(this.#updates.rect);
+    this.#updates = null;
+  }
+
+  #setRect(rect) {
```
Comment (Snuffleupagus) [inline src/display/annotation_layer.js:228]: Maybe update this name as well, to avoid accidental future use elsewhere?
```suggestion
  #setEditedRect(rect) {
```

**Review by Snuffleupagus — APPROVED** [review]
r=me, trusting that our test-coverage is sufficient to catch any edge-case problems; thank you!
